backend/molgenis-emx2-sql/src/main/resources/org/molgenis/emx2/sql/migration1.sql

```diff
@@ -0,0 +1,4 @@
+TRUNCATE "MOLGENIS"."version_metadata";
+ALTER TABLE "MOLGENIS"."version_metadata" DROP COLUMN "version";
+ALTER TABLE "MOLGENIS"."version_metadata"
+    ADD COLUMN "version" INTEGER NOT NULL;
```

Comment:
should probably add a comment ( according to some template defined in the dev docs) to each migration explaining why it is needed and what version is ( or add some kind of file naming schema to the docs. Maybe use a migration folder or a folder per version of something. This could also be a way of avoiding the whole if , if , if and setting a version in code , the app could for example scan a folder and determine the migrations that need to be run ( based on file or folder naming), but then again that would maybe make is 'harder' to do code migrations ( instead of sql)

Reply:
tempting. if we would have all migrations in sql, then simply 'migration1.sql', 'migration2.sql' etc would work. However I think migrations may take different forms, sometimes sql, sometimes java, maybe even something else. Therefore I believe we should document this in the Migrations.java as part of their log messages for a step.